pkg/server/endpoints/entryfetcher_test.go

```diff
@@ -0,0 +1,199 @@
+package endpoints
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"net/url"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/andres-erbsen/clock"
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/server/api"
+	"github.com/spiffe/spire/pkg/server/plugin/datastore"
+	"github.com/spiffe/spire/proto/spire/common"
+	"github.com/spiffe/spire/proto/spire/types"
+	"github.com/spiffe/spire/test/fakes/fakedatastore"
+	"github.com/spiffe/spire/test/fakes/fakemetrics"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	spiffeScheme = "spiffe"
+	trustDomain  = "example.org"
```

Comment:
I'd suggest leveraging the go-spiffe v2 types here. It removes a lot of the boilerplate.
```suggestion
	trustDomain  = spiffeid.RequireTrustDomainFromString("example.org")
```

Reply:
Oh yeah that makes things a lot cleaner!

_finally explores go-spiffe v2 types_